### DISCUSSION TITLE
SE10000H & 2-SE Energy Bank 9.7k: no batteries found via modbus

### DISCUSSION THREAD

**Discussion by cpaquette63** [WillCodeForCats/solaredge-modbus-multi#225] [General]
Just getting started and installed your integration.  I have enabled Modbus and I am receiving I1, M1, & M2 values but no B1 or B2.  

My Inverter is a SE10000H with backup interface and 2 Energy Bank 9.7k batteries.

Any idea as to what I may be doing wrong?

Thanks!

**Comment by metakevin** [comment]
I have the same setup and the same issue.  If I turn on debug logging I get this:

```
2023-01-09 23:08:44.051 DEBUG (SyncWorker_4) [custom_components.solaredge_modbus_multi.hub] Inverter 1 battery 1: Exception Response(131, 3, IllegalAddress)
2023-01-09 23:08:44.486 DEBUG (SyncWorker_3) [custom_components.solaredge_modbus_multi.hub] Inverter 1 battery 2: Exception Response(131, 3, IllegalAddress)
```
(full startup log [here](https://gist.github.com/metakevin/aef6711a762331f3e14ac8639b08efe2) if it's interesting)

Using the solaredge_modbus library directly, I get a similar response, so I don't think this is related to solaredge-modbus-multi specifically, but maybe someone who reads this will have a suggestion.

```
>>> import solaredge_modbus
>>> master = solaredge_modbus.Inverter(host="192.168.13.186", port=1502)

>>> master.read_all()
{'c_id': 'SunS', 'c_did': 1, 'c_length': 65, 'c_manufacturer': 'SolarEdge', 'c_model': 'SE10000H-USSNBBL14', 'c_version': '0004.0017.0046', 'c_serialnumber': '740E6E2E', 'c_deviceaddress': 1, 'c_sunspec_did': 101, 'c_sunspec_length': 50, 'current': 41, 'l1_current': 41, 'l2_current': 0, 'l3_current': 0, 'current_scale': -2, 'l1_voltage': 2474, 'l2_voltage': 0, 'l3_voltage': 0, 'l1n_voltage': 2474, 'l2n_voltage': 0, 'l3n_voltage': 0, 'voltage_scale': -1, 'power_ac': -4055, 'power_ac_scale': -3, 'frequency': 59998, 'frequency_scale': -3, 'power_apparent': 10198, 'power_apparent_scale': -2, 'power_reactive': 10190, 'power_reactive_scale': -2, 'power_factor': -3976, 'power_factor_scale': -3, 'energy_total': 22115, 'energy_total_scale': 0, 'current_dc': 65527, 'current_dc_scale': -3, 'voltage_dc': 4361, 'voltage_dc_scale': -1, 'power_dc': -4117, 'power_dc_scale': -3, 'temperature': 2704, 'temperature_scale': -2, 'status': 1, 'vendor_status': 0}

>>> master.batteries()
{'Battery1': Battery1(192.168.13.186:1502, connectionType.TCP: timeout=1, retries=3, unit=0x1), 'Battery2': Battery2(192.168.13.186:1502, connectionType.TCP: timeout=1, retries=3, unit=0x1)}
>>> b1 = master.batteries()["Battery1"]
>>> b1.read_all()
{}

>>> master._read_holding_registers(0xe140, 1)
>>> master._read_holding_registers(0xe140, 1)
>>> master._read_holding_registers(0xe100, 1)
>>> master._read_holding_registers(0xe100, 1)
```

The inverter knows about the batteries:

![image](https://user-images.githubusercontent.com/1704224/211460774-fc436aa6-76fe-47a1-bd3a-81cd20da3f05.png)


I am wondering if this is related to the control interface between the inverter and batteries: it's wireless, not RS-485.  Solaredge calls this EnergyNet in some of the documentation I've found.

I do see an optional RS-485 connection available on the batteries but hopefully getting battery status into HA doesn't require additional hardware...

**Reply by WillCodeForCats** [reply]
IllegalAddress means the inverter does not support batteries over the modbus interface. This is almost universally the case for any "batteries don't show up" problem and it can only be resolved by SolarEdge support.

Some people have been successful in convincing SolarEdge to enable this functionality for them: https://github.com/WillCodeForCats/solaredge-modbus-multi/discussions/63#discussioncomment-4510743

Others have not had success with SolarEdge: https://github.com/WillCodeForCats/solaredge-modbus-multi/discussions/63#discussioncomment-4500443

All of it is discussed in discussion https://github.com/WillCodeForCats/solaredge-modbus-multi/discussions/63

As far as the integration goes there's nothing that can be done in SetApp or in the integration: if you're not lucky enough to have an inverter that supports battery data over modbus/tcp out of the box you will have to try your luck with convincing SolarEdge support to push an update to your inverter to enable it.

**Reply by metakevin** [reply]
Thanks!  My setup is brand new and the installers are still wrapping things up.  I'll see if I can get them to follow up on this.

**Reply by cpaquette63** [reply]
metakevin: if your installer is able to resolve please provide details.  My system went live at the end of December so my installers are done locally but are still supporting.  Unfortunately, I've not been able to get admin access (as of yet!!) so I can go play around so still relying on the installers.

Thanks Seth and Kevin!  At least it is not something I'm doing on my end.
